### PR TITLE
Feature/cdpt 2075 search bar errors

### PR DESCRIPTION
**Pull request by EmilyHazlehurst**

### CODE REVIEW

File: public/app/themes/justice/search.php [hunk 1]
```diff
@@ -71,14 +71,14 @@
...
 
     // If the post type is document get the format and filesize
     if ($document->verify_post_type($postId)) {
         $upload_dir = $document->document_upload_dir();
         $year_month = str_replace('-', '/', substr($result->post_date, 0, 7));
         $format = $document->get_file_type($postId);
-        $document_url = "{$upload_dir}/{$year_month}/{$result->post_title}{$format}";
+        $document_url = "{$upload_dir}/{$year_month}/{$result->slug}{$format}";
         if (file_exists($document_url)) {
             $filesize = size_format(wp_filesize($document_url));
```
Comment (EarthlingDavey) [inline public/app/themes/justice/search.php:83]: While this code wasn't introduce in this PR, I've spotted something that might be a bug.

As we are offloading documents to S3, they won't be on the filesystem - I think checking the file size like this won't work.

However, WP Offload Media does save the file size in the database, there is a function that you can either use directly or copy. This screenshot should give an overview of the function and database structure.

image

File: public/app/themes/justice/search.php [hunk 2]
```diff
@@ -98,11 +98,29 @@
...
 $pagination = $results->pagination();
+$didYouMean = [];
 $suggestedTerm = relevanssi_premium_generate_suggestion($query);
-$didYouMean = [
-    'term' => $suggestedTerm,
-    'url' => `search/${suggestedTerm}`
-];
+// If the term used is valid relevanssi_premium_generate_suggestion returns true
+// We don't need to display anything if this is the case
+if ($suggestedTerm && $suggestedTerm !== true) {
```
Comment (EarthlingDavey) [inline public/app/themes/justice/search.php:105]: `if ($suggestedTerm !== true) {` might be ok here.

File: public/app/themes/justice/search.php
Comment (EarthlingDavey) [inline public/app/themes/justice/search.php:1]: `get_footer();` should be at the end of the page. For example some script tags are added by `get_footer` and they are expected to be just before the closing `body` tag.

**Review by EarthlingDavey — APPROVED** [review]
Nice work. I've run it on localhost and it's working well!

I've written some suggestions and feedback - nothing major :)

File: public/app/frontend/src/components/sidebar-block/sidebar-block.html.twig [hunk 3]
```diff
@@ -97,22 +97,31 @@ example:
...
-                                options: field.options,
-                                disabled: field.disabled,
-                                error: field.error,
-                            } %}
-                        </div>
+                        {% if field.options %}
+                            <div class="sidebar-block__search-filter-field">
+                                {% include '@components/selection-input/selection-input.html.twig' with {
+                                    type: field.type,
+                                    direction: field.direction,
```
[the remark this answers is not in the document]
Comment (EmilyHazlehurst) [reply public/app/frontend/src/components/sidebar-block/sidebar-block.html.twig:112]: @EarthlingDavey I didn't know twig allowed the spread operator! Looks like it was added fairly recently. Really good to know, thank you :)

File: public/app/frontend/src/components/sidebar-block/sidebar-block.html.twig [hunk 4]
```diff
@@ -97,22 +97,31 @@ example:
...
         {% elseif variant == 'search-filter' %}
-            <form class="sidebar-block__search-filter">
-                <p class="sidebar_block__search-filter-subtitle">{{ subtitle }}</p>
+            <form
+                class="sidebar-block__search-filter {{ noQuery ? " sidebar-block__search-filter--disabled" : "" }}"
+                {{ noQuery ? " aria-describedby='no-query-hint'" : "" }}>
+                <p class="sidebar_block__search-filter-subtitle" {{ noQuery ? " id='no-query-hint'" : "" }}>
+                    {{ noQuery ? 'Please enter a search query to use filters' : subtitle }}
+                </p>
                 <div class="sidebar-block__search-filter-fields">
```
[the remark this answers is not in the document]
Comment (EarthlingDavey) [reply public/app/frontend/src/components/sidebar-block/sidebar-block.html.twig:106]: I didn't realise you already used fieldsets, yeah let's not nest them 😁

**Pull request merged**